### ISSUE TITLE
Bug: app still quits unexpectedly and doesn't relaunch

### ISSUE THREAD

**Issue by bdruth** [leits/MeetingBar#692]
### macOS version

14.3.1

### MeetingBar version

4.7.1

### Installation source

Homebrew

### Calendars provider

Google Calendar

### Bug description

Seems like https://github.com/leits/MeetingBar/issues/463 - in a nutshell, I regularly will notice that meetingbar just isn't running ... launch it and everything's good. Seems to just crash in the background or something.

### Expected behavior

Stay running ... if crashes are inevitable, then at least relaunch automatically

**Comment by mega-joe** [comment]
I'm having the same issue. I notice it after waking the Mac from sleep or "docking"/"undocking" with my monitor and other ports. If I can provide anything, please ask.

**Comment by stunningpixels** [comment]
Also happens for me, was using an old version and recently upgraded to the latest but it's still happening. Quits a few times a day

**Comment by mikedloss** [comment]
(this might be related to https://github.com/leits/MeetingBar/issues/656)

**Comment by BlaineEXE** [comment]
Same is happening to me. It happens every 7-10 day's I'd guess.

**Comment by stenalpjolly** [comment]
This is happening to me as well. 

Apple M3 Pro
macOS version: 14.6.1 (23G93)
MeetingBar version: 4.10.0

Any way to get the logs for more details?

**Comment by mikedloss** [comment]
Unsure if this will ever get fixed or not, but I've since switched to using Raycast for this. The "Calendar" extension lets you sync with calendars, and there's an option to show events in the menu bar.

**Comment by darthwalsh** [comment]
It would be awesome if the app could relaunch!

But for anybody on this thread seeing crashes, we could also help the maintainers to figure out the root cause of the app crash.

If you open Console.app > Diagnostic Reports > search `MeetingBar` you will probably find a relevant crash.
In the bottom details planel, if you search `""termination"` you might find some more details:

```json
  "termination": {
    "flags": 2,
    "code": 14123288431434181290,
    "namespace": "PORT_SPACE",
    "indicator": "(Limit 305834 ports) Exceeded system-wide per-process Port Limit"
  }
```

which means that issue #656 looks like the root cause for my crash, and I can go upvote that too!

**Comment by stenalpjolly** [comment]
Same root cause for me as well

<img width="787" alt="Image" src="https://github.com/user-attachments/assets/13e63df0-d4a7-4297-aa81-a04ee29cabed" />


```
Exception Type:        EXC_RESOURCE (SIGKILL)
Exception Codes:       0x000000000004aaaa, 0x0000000000000000

Termination Reason:    Namespace PORT_SPACE, Code 14123288431434181290 (Limit 305834 ports) Exceeded system-wide per-process Port Limit
```

**Comment by mikedloss** [comment]
not to detract from the issue here, but if you use Raycast it has this feature built in now, no need for meeting bar at all 😬

**Comment by mega-joe** [comment]
> not to detract from the issue here, but if you use Raycast it has this feature built in now, no need for meeting bar at all 😬

Yep. As a different alternative, I switched to "In Your Face" which has worked perfectly. Too many crashes and memory issues to rely on MeetingBar, unfortunately.

**Comment by leits** [comment]
Hi folks,
Thanks again for reporting this performance issue. I’ve just published a 4.11 beta release with a number of under-the-hood performance optimizations in MeetingBar. You can grab it here:
https://github.com/leits/MeetingBar/releases/tag/v4.11.beta 

Once you’ve had a chance to try it, could you please let me know whether you still see the issues? Any logs, profiled timings, or specific repro steps you can share will be super helpful. Thanks!

**Comment by leits** [comment]
I have finalized the release and am gradually rolling it out, you can try it here:
https://github.com/leits/MeetingBar/releases/tag/v4.11.2

**Comment by ialexryan** [comment]
Hi @leits would you mind reopening this issue? Unfortunately even on 4.11.6 I'm still finding MeetingBar not running multiple times a day. (Second report [here](https://github.com/leits/MeetingBar/issues/863).)
